src/language/rule/rule.ts

```diff
@@ -18,6 +18,53 @@
 import * as ts from "typescript";
 import {RuleWalker} from "../walker/ruleWalker";
 
+export interface IRuleMetadata {
+   /**
+    * The kebab-case name of the rule.
+    */
+    ruleName: string;
+
+   /**
+    * The type of the rule - its overall purpose
+    */
+    type: RuleType;
+
+   /**
+    * A short, one line description of what the rule does.
+    */
+    description: string;
+
+   /**
+    * More elaborate details about the rule.
+    */
+    descriptionDetails?: string;
+
+   /**
+    * An explanation of the available options for the rule.
+    */
+    optionsDescription?: string;
+
+   /**
+    * Schema of the options the rule accepts.
+    * The first boolean for whether the rule is enabled or not is already implied.
+    * This field describes the options after that boolean.
+    * If null, this rule has no options and is not configurable.
+    */
+    options: any;
+
+   /**
+    * Examples of what a standard config for the rule might look like.
+    */
+    optionExamples?: string[];
+
+   /**
+    * An explanation of why the rule is useful.
+    */
+    rationale?: string;
+}
+
+export type RuleType = "functionality" | "maintainability" | "style" | "typescript";
```

Comment:
I'm not sure we use string literal union types anywhere else, and this might break things for 1.7 consumers. I feel okay with that at this point because 1.8 has been out for awhile, but we could use a different tool instead of string literal union types if we need to